clearnode/nitrolite/State.md

```diff
@@ -0,0 +1,155 @@
+
+-----
+
+# Yellow Network Definition
+
+Description generated from the State spec by Gemini
+
+Documentation
+
+## 1\. Overview
+
+The Yellow Network is a high-performance overlay mesh network designed to operate across multiple blockchains. It provides users with a unified account, identified by a unique `UserID`, which aggregates their asset holdings across all integrated chains into a single `CommonState`. This architecture enables high-speed, low-cost off-chain transactions, deep liquidity, and a seamless cross-chain user experience.
+
+All state transitions within the network are authorized by users and validated by a dynamic, weighted quorum of **Ledger Nodes**. A master smart contract on Ethereum maintains the network configuration, while on-chain **Custody** and **Adjudicator** contracts on each supported chain handle deposits, withdrawals, and dispute resolution.
+
+## 2\. Core Concepts
+
+### Actors
+
+  * **User**: The owner of the funds and the `CommonState`, identified by a unique Yellow Network `UserID`. The user authorizes all actions by signing intents or states with their primary key or a delegated session key.
+  * **Ledger Nodes (Clearnode)**: A permissioned set of off-chain nodes responsible for validating user intents, issuing new states, reaching consensus, and attesting to state validity with their weighted signatures. Their registry and weights are managed by the Adjudicator contract.
+  * **Master Registry Contract (on Ethereum)**: The primary on-chain contract defining the network's configuration, including approved ledger node keys, supported blockchains and the addresses of their respective Custody and Adjudicator contracts.
+  * **Custody Contract (on each chain)**: Holds the pooled funds for all users on a specific blockchain, processes deposits, and executes withdrawals upon receiving valid proofs.
+  * **Adjudicator Contract (on each chain)**: A companion to the Custody contract that validates state proofs. It verifies signatures, and resolves disputes by computationally verifying forced state transitions.
+
+### Core Data Structures
+
+#### `CommonState`
+
+This is the master object representing a user's account, bundling the state data with the necessary signatures to prove its validity.
```

Comment:
Some renaming suggestions:

- `CommonState` -> `State` or `Bundle`
- `UnsignedCommonState` -> `UnsignedState`
- The field `StateData` -> one of `Data`, `Payload`, or `RawBytes`
-